From 01eaf1371518eaf29d9197b2f3cc7923980dc849 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Mon, 18 Jan 2016 16:32:49 +0100
Subject: [PATCH] Fix NULL pointer reference when using basic match

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1298933

commit 4b45abd1f000f62df9b624e393d9f3d62751a266
Author: Stephen Hemminger <stephen.hemminger@vyatta.com>
Date:   Thu Jul 29 18:03:35 2010 -0700

    Fix NULL pointer reference when using basic match

    If basic match has no tree of matches underneath
    then print_ematch would core dump.
---
 tc/m_ematch.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/tc/m_ematch.c b/tc/m_ematch.c
index 7f79a065a1d76..4c3acf82b470a 100644
--- a/tc/m_ematch.c
+++ b/tc/m_ematch.c
@@ -449,12 +449,14 @@ static int print_ematch_list(FILE *fd, struct tcf_ematch_tree_hdr *hdr,
 	if (tb == NULL)
 		return -1;
 
-	if (parse_rtattr_nested(tb, hdr->nmatches, rta) < 0)
-		goto errout;
+	if (hdr->nmatches > 0) {
+		if (parse_rtattr_nested(tb, hdr->nmatches, rta) < 0)
+			goto errout;
 
-	fprintf(fd, "\n  ");
-	if (print_ematch_seq(fd, tb, 1, 1) < 0)
-		goto errout;
+		fprintf(fd, "\n  ");
+		if (print_ematch_seq(fd, tb, 1, 1) < 0)
+			goto errout;
+	}
 
 	err = 0;
 errout:
-- 
2.6.4

